### ISSUE TITLE
ZeroDivisionError

### ISSUE THREAD

**Issue by stevenreinisch** [nakiostudio/xcov#8]
Hello,

I got a ZeroDivisionError. The crash report has been submitted. The number is dbaf0b8d5f8062731ccd1b662620d467.

best,

steven

**Comment by nakiostudio** [comment]
Hi @stevenreinisch. Thanks for the feedback but that report is submitted to Fastlane, not to me. Could you please give me more details? logs?

Thanks :+1:

**Comment by stevenreinisch** [comment]
Hi,

I ran `scan` on my project and then `xcov`. It crashed with the following stack trace:

```
.rvm/gems/ruby-2.2.1@global/gems/xcov-0.2/lib/xcov/model/report.rb:23:in `/': [!] divided by 0 (ZeroDivisionError)
	from .rvm/gems/ruby-2.2.1@global/gems/xcov-0.2/lib/xcov/model/report.rb:23:in `average_coverage'
	from .rvm/gems/ruby-2.2.1@global/gems/xcov-0.2/lib/xcov/model/report.rb:12:in `initialize'
	from .rvm/gems/ruby-2.2.1@global/gems/xcov-0.2/lib/xcov/model/report.rb:49:in `new'
	from .rvm/gems/ruby-2.2.1@global/gems/xcov-0.2/lib/xcov/model/report.rb:49:in `map'
	from .rvm/gems/ruby-2.2.1@global/gems/xcov-0.2/lib/xcov/runner.rb:53:in `generate_xcov_report'
	from .rvm/gems/ruby-2.2.1@global/gems/xcov-0.2/lib/xcov/runner.rb:14:in `run'
	from .rvm/gems/ruby-2.2.1@global/gems/xcov-0.2/lib/xcov/manager.rb:9:in `work'
	from .rvm/gems/ruby-2.2.1@global/gems/xcov-0.2/lib/xcov/commands_generator.rb:41:in `block (2 levels) in run'
	from .rvm/gems/ruby-2.2.1@global/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from .rvm/gems/ruby-2.2.1@global/gems/commander-4.3.5/lib/commander/command.rb:178:in `call'
	from .rvm/gems/ruby-2.2.1@global/gems/commander-4.3.5/lib/commander/command.rb:153:in `run'
	from .rvm/gems/ruby-2.2.1@global/gems/commander-4.3.5/lib/commander/runner.rb:428:in `run_active_command'
	from .rvm/gems/ruby-2.2.1/gems/fastlane_core-0.36.6/lib/fastlane_core/ui/fastlane_runner.rb:23:in `run!'
	from .rvm/gems/ruby-2.2.1@global/gems/commander-4.3.5/lib/commander/delegates.rb:15:in `run!'
	from .rvm/gems/ruby-2.2.1@global/gems/xcov-0.2/lib/xcov/commands_generator.rb:47:in `run'
	from .rvm/gems/ruby-2.2.1@global/gems/xcov-0.2/lib/xcov/commands_generator.rb:15:in `start'
	from .rvm/gems/ruby-2.2.1@global/gems/xcov-0.2/bin/xcov:6:in `<top (required)>'
	from .rvm/rubies/ruby-2.2.1/bin/xcov:23:in `load'
	from .rvm/rubies/ruby-2.2.1/bin/xcov:23:in `<main>'
	from .rvm/gems/ruby-2.2.1/bin/ruby_executable_hooks:15:in `eval'
	from .rvm/gems/ruby-2.2.1/bin/ruby_executable_hooks:15:in `<main>'
```

Looking at the code, I guess that there are no "targets" here: https://github.com/nakiostudio/xcov/blob/master/lib/xcov/model/report.rb#L45

I have a test target in my project. Why are there none found my project? My test target contains a whitespace in its name.

Thanks,

Steven

**Comment by nakiostudio** [comment]
Hi again @stevenreinisch. I think I know what's the problem, I will release a hot-fix asap. About your particular case the target name shouldn't matter but I will dig into that. Thanks for reporting :+1: .

**Comment by stevenreinisch** [comment]
Thanks man!

**Comment by nakiostudio** [comment]
Hi @stevenreinisch :wave: 

I've just released the version 0.4 fixing this problem and enabling the coverage reports for test targets. To do so you just need to add the `include_test_targets ` flag when launching xcov. Example: 

```
xcov -w LystSDK.xcworkspace -s LystSDK -o xcov_output --include_test_targets
```

It'd be great if you can confirm whether this update fixes your problems.

Thank you!

**Comment by stevenreinisch** [comment]
Hi @nakiostudio,

the 0.4 fixes the issue. But :), I had to remove all DerivedData. If not, xcov took an older .xccoverage file - not the most recent one.

Thanks for your efforts!

**Comment by nakiostudio** [comment]
I would like into that @stevenreinisch Thanks!

**Comment by nakiostudio** [comment]
Hi @stevenreinisch. I've released the version 0.5 which sorts out the problem picking the right `.xccoverage` file :+1:
